app/src/main/kotlin/at/bitfire/davdroid/ui/setup/LoginActivity.kt

```diff
@@ -23,6 +23,33 @@ import javax.inject.Inject
 @AndroidEntryPoint
 class LoginActivity @Inject constructor(): AppCompatActivity() {
 
+    @Inject lateinit var loginTypesProvider: LoginTypesProvider
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val initialLoginType = loginTypesProvider.intentToInitialLoginType(intent)
+        val skipLoginTypePage = initialLoginType != loginTypesProvider.defaultLoginType
```

Comment:
The `defaultLoginType` is `UrlLogin`. So this means when

1. the Intent's URL is `https://example.com/` →
2. `initialLoginType == LoginUrl` (login with URL; except when Nextcloud Flow extra is set) →
3. `skipLoginTypePage` is always `false`.

However, it's clear that "Login with URL" should be chosen, so shouldn't it skip the login type too and go directly to "Login with URL"?

Reply:
Ah yes. You are completely right. I guess I will turn it back to using a pair of values. It seems it makes the most sense after all.

Reply:
Maybe even a more typed value, like a data class with two named properties? `Pair` is comfortable, but on the other side a bit dirty.

Reply:
Right. For lack of a better name I called the data class `LoginAction`.